diff -ur perl-5.8.7/Makefile.micro perl-5.8.7-magicbox/Makefile.micro
--- perl-5.8.7/Makefile.micro	2004-07-02 17:03:34.000000000 +0200
+++ perl-5.8.7-magicbox/Makefile.micro	2006-03-03 00:36:17.000000000 +0100
@@ -1,8 +1,9 @@
+CC = powerpc-linux-gcc
 LD = $(CC)
-DEFINES = -DPERL_CORE -DPERL_MICRO -DSTANDARD_C -DPERL_USE_SAFE_PUTENV
-OPTIMIZE = 
+DEFINES = -DPERL_CORE -DSTANDARD_C -DPERL_USE_SAFE_PUTENV
+OPTIMIZE = -Os
 CFLAGS = $(DEFINES) $(OPTIMIZE)
-LIBS = -lm
+LIBS = -lm -lcrypt
 _O = .o
 ENV = env
 
@@ -21,7 +22,7 @@
 microperl:	$(O)
 	$(LD) -o $@ $(O) $(LIBS)
 
-H = av.h uconfig.h cop.h cv.h embed.h embedvar.h form.h gv.h handy.h \
+H = av.h config.h cop.h cv.h embed.h embedvar.h form.h gv.h handy.h \
 	hv.h intrpvar.h iperlsys.h mg.h op.h opcode.h opnames.h pad.h \
 	patchlevel.h perl.h perlsdio.h perlvars.h perly.h pp.h \
 	pp_proto.h proto.h reentr.h regexp.h scope.h sv.h thrdvar.h \
@@ -34,10 +35,10 @@
 
 distclean:	clean
 
-# The microconfiguration.
+# The microconfiguration.  NOT.
 
 regen_uconfig:
-	$(ENV) CONFIG_SH=uconfig.sh CONFIG_H=uconfig.h sh ./config_h.SH
+	$(ENV) CONFIG_SH=config.sh CONFIG_H=config.h sh ./config_h.SH
 
 # Do not regenerate perly.c and perly.h.
 
